derp/derp_server.go

```diff
@@ -343,6 +350,9 @@ func (s *Server) registerClient(c *sclient) {
 	if _, ok := s.clientsMesh[c.key]; !ok {
 		s.clientsMesh[c.key] = nil // just for varz of total users in cluster
 	}
+	if !c.remoteIPPort.IsZero() {
```

Comment:
you could also just skip the IsZero check. not the end of the world if you add/remove useless/redundant keys under the key's zero value. Nobody will look it up by that, so it's harmless.